xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -606,6 +610,82 @@ BuildObject(CFileItem&                    item,
         }
     }
 
+    //Add external subtitle
+    if (upnp_server && item.IsVideo()) //only if video file
+    {
+        // find any available external subtitles
+        std::vector<std::string> filenames;
+        std::vector<std::string> subtitles;
+        CUtil::ScanForExternalSubtitles(file_path.GetChars(), filenames);
+
+        NPT_String ext;
+        for (int i = 0; i < filenames.size(); i++)
+        {
+            ext = URIUtils::GetExtension(filenames[i]).c_str();
+            ext.TrimLeft('.');
+            ext = ext.ToLowercase();
+			/* Hardcoded check for extension is not the best way, but it can't be allowed to pass all
+			   subtitle extension (ex. rar or zip). There are the most popular extensions support by UPnP devices.*/
+            if (ext == "txt" || ext == "srt" || ext == "ssa" || ext == "ass" || ext == "sub" || ext == "smi")
+            {
+                subtitles.push_back(filenames[i]);
+            }
+        }
+
+        std::string subtitlePath = "";
+
+        if (subtitles.size() == 1)
+        {
+            subtitlePath = subtitles[0];
+        }
+        else if (!subtitles.empty())
+        {
+            /* trying to find subtitle with prefered language settings */
+            std::string preferedLanguage = (CSettings::Get().GetSetting("locale.subtitlelanguage"))->ToString();
```

Comment:
This logic probably already exists somewhere (probably in DVDPlayer) so would be nice if it could be refactored so it doesn't have to be duplicated.